pkg/exthttp/middleware.go

```diff
@@ -0,0 +1,87 @@
+package exthttp
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type ServerMetrics struct {
```

Comment:
This is bit changed version:
* No interface, no nop (why it was needed even?)
* InFlight gauge
* handler reuses Wrap technique
* No method just function for new handler, no need to combine those together IMO.